### PR TITLE
Use Ruby 3.1.3

### LINKED ISSUE #97: [TASK] Upgrade the app to use Ruby 3.1
**Problem:**

The app is currently using Ruby 3.0 which is not compatible with the Heroku-22 stack.

**Acceptance criteria:**

The app runs in Ruby 3.1.

**I will abide by the [code of conduct](https://github.com/fastruby/skunk.fyi/blob/main/CODE_OF_CONDUCT.md)**

### PR DESCRIPTION
**Pull request by arielj**
This PR fixes #97 

I upgraded the Ruby version from 3.0.2 to 3.1.3. I upgraded to the latest 6.1 to make sure it works with ruby 3.1. I also changed the dual boot to be between 6.1 and 7.0 (it was using the main branch so we were dual booting between 6.1 and 7.1-alpha).

I also had to change some code because upgrading to Ruby 3.1 required upgrading standardrb and that generated a few new offences.

I will abide by the [code of conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: .github/workflows/skunk.yml [hunk 1]
```diff
@@ -27,7 +27,10 @@ jobs:
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: "3.0.2"
+          ruby-version: "3.1.3"
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
Comment (arielj) [inline .github/workflows/skunk.yml:33]: the 3.1.3 image installs node 18 by default but webpacker needs node 16 to work

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -29,7 +29,10 @@ jobs:
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: "3.0.2"
+          ruby-version: "3.1.3"
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
Comment (arielj) [inline .github/workflows/tests.yml:35]: the 3.1.3 image installs node 18 by default but webpacker needs node 16 to work

File: Gemfile [hunk 3]
```diff
@@ -5,11 +5,11 @@ end
 source "https://rubygems.org"
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby "3.0.2"
+ruby "3.1.3"
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 if next?
-  gem "rails", github: "rails/rails", branch: "main"
+  gem "rails", "~> 7.0.0"
```
Comment (arielj) [inline Gemfile:12]: if I kept `main` here it was going to use 7.1-alpha instead of 7.0

File: Gemfile.lock [hunk 4]
```diff
@@ -184,21 +197,21 @@ GEM
       kwalify (~> 0.7.0)
       parser (~> 3.0.0)
       rainbow (>= 2.0, < 4.0)
-    regexp_parser (2.1.1)
+    regexp_parser (2.7.0)
     require_all (3.0.0)
     rexml (3.2.5)
-    rubocop (1.20.0)
+    rubocop (1.24.1)
```
Comment (arielj) [inline Gemfile.lock:203]: standard/rubocop had to be updated for ruby 3.1

File: app/controllers/reports_controller.rb [hunk 5]
```diff
@@ -11,7 +11,7 @@ def create
     project = find_project(input)
 
     unless entries.is_a? Array
-      head 400
+      head :bad_request
```
Comment (arielj) [inline app/controllers/reports_controller.rb:14]: from this file down, these are fixes for standard/rubocop offences

**Comment by rishijain** [comment]
so we will be doing a different PR for it to upgrade it to rails 7.0 @arielj ?

**Review by rishijain — APPROVED** [review]

**Review by mateusdeap — APPROVED** [review]

**Review by mateusdeap — APPROVED** [review]

**Pull request merged**